src/snowflake/cli/_app/snow_connector.py

```diff
@@ -46,6 +44,22 @@
 ENCRYPTED_PKCS8_PK_HEADER = b"-----BEGIN ENCRYPTED PRIVATE KEY-----"
 UNENCRYPTED_PKCS8_PK_HEADER = b"-----BEGIN PRIVATE KEY-----"
 
+# connection keys that can be set using SNOWFLAKE_* env vars
+SUPPORTED_ENV_OVERRIDES = [
```

Comment:
This is necessary because before the possible overrides was implicit in the keys of `_ConnectionContext._collect_not_empty_connection_attributes`, which was a misnomer because it always contained the same keys whether or not they had values. By modifying it to the (assumed) original intention, we needed a list of keys that can be grabbed from `os.environ` somewhere (this is a subset of all keys in `ConnectionContext`).